ui/admin-portal/src/app/components/settings/users/create-update-user/create-update-user.component.spec.ts

```diff
@@ -0,0 +1,198 @@
+    component.ngOnInit();
+    expect(component.userForm.value.email).toBe('john.doe@example.com');
+    expect(component.userForm.value.username).toBe('mockuser');
+    expect(component.userForm.value.firstName).toBe('John');
+    expect(component.userForm.value.lastName).toBe('Doe');
+  });
+
+  it('should call create method on save for new user', () => {
+    component.user = null; // Ensure it's a new user
+    component.ngOnInit();
+    component.userForm.setValue({
+      email: 'newuser@example.com',
+      username: 'newuser',
+      firstName: 'New',
+      lastName: 'User',
+      partnerId: 1,
+      status: 'active',
+      role: 'user',
+      jobCreator: false,
+      approverId: null,
+      purpose: '',
+      sourceCountries: [],
+      readOnly: false,
+      usingMfa: true,
+      password: 'password'
+    });
+    component.onSave();
+    expect(userServiceSpy.create).toHaveBeenCalled();
+  });
+
+  it('should call update method on save for existing user', () => {
+    component.user = updatedMockUser;
+    component.ngOnInit();
+    component.onSave();
+    expect(userServiceSpy.update).toHaveBeenCalled();
+  });
+
+  it('should display an error message on save failure', () => {
+    userServiceSpy.create.and.returnValue(throwError('error'));
+    component.user = null; // Ensure it's a new user
+    component.ngOnInit();
+    component.userForm.setValue({
+      email: 'newuser@example.com',
+      username: 'newuser',
+      firstName: 'New',
+      lastName: 'User',
+      partnerId: 1,
+      status: 'active',
+      role: 'user',
+      jobCreator: false,
+      approverId: null,
+      purpose: '',
+      sourceCountries: [],
+      readOnly: false,
+      usingMfa: true,
+      password: 'password'
+    });
+    component.onSave();
+    expect(component.error).toBe('error');
+  });
+
+  it('should filter role options based on logged-in role', () => {
+    authServiceSpy.getLoggedInRole.and.returnValue(Role.admin);
+    component.ngOnInit();
+    expect(component.roleOptions.some(role => role.key === 'systemadmin')).toBeFalse();
+  });
+
+  it('should correctly identify if the component is in create mode', () => {
+    component.user = null; // Ensure it's a new user
+    expect(component.create).toBeTrue();
+    component.user = mockUser;
+    expect(component.create).toBeFalse();
+  });
+
+  it('should close the modal on successful save', () => {
+    const activeModal = TestBed.inject(NgbActiveModal);
+    spyOn(activeModal, 'close');
+    component.user = null; // Ensure it's a new user
+    component.ngOnInit();
+    component.userForm.setValue({
+      email: 'newuser@example.com',
+      username: 'newuser',
+      firstName: 'New',
+      lastName: 'User',
+      partnerId: 1,
+      status: 'active',
+      role: 'user',
+      jobCreator: false,
+      approverId: null,
+      purpose: '',
+      sourceCountries: [],
+      readOnly: false,
+      usingMfa: true,
+      password: 'password'
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected credentials hardcoded in your code, which might allow unauthorized users to access your account. We recommend you to store your credentials outside of the code in a configuration file, a database, or a management service for secrets and retrieve from it. For more information, see [CWE-259](https://cwe.mitre.org/data/definitions/259.html) and [CWE-798](https://cwe.mitre.org/data/definitions/798.html).



Suggested remediation:
Do not hardcode password or access key. Instead, try to fetch it from environment to deny unauthorized users to access your account.
```
@@ -185,3 +185,3 @@
       usingMfa: true,
-      password: &#39;password&#39;
+      password: process.env.ENV_VARIABLE
     });

```